ibl_pipeline/job/patch.py

```diff
@@ -0,0 +1,349 @@
+    table_parent=''         : varchar(128)  # only applicable to part table
+    """
+
+    @classmethod
+    def _insert_package_tables(self, table_list, table_type):
+        for itable, table in enumerate(table_list[::-1]):
+            table_obj = eval(table)
+            table_key = dict(full_table_name=table_obj.full_table_name)
+
+            if self & table_key:
+                dj.Table._update(self & table_key, 'table_order', itable)
+            else:
+                self.insert1(dict(
+                    **table_key,
+                    table_class=table,
+                    table_order=itable,
+                    table_order_category=table_type,
+                    table_label='auto' if issubclass(table_obj,
+                                                     (dj.Imported, dj.Computed))
+                                else 'part',
+                    table_parent=eval(re.match('(^.*)\..*$', table).group(1)).full_table_name
+                                 if issubclass(table_obj, dj.Part) else None))
+
+    def _insert_virtual_tables(self):
+        # insert virtual modules tables in order
+        virtuals = Graph(behavior.TrialSet()).get_table_list(virtual_only=True) + \
+            Graph(ephys.DefaultCluster()).get_table_list(virtual_only=True) + \
+            Graph(behavior_analyses.BehavioralSummaryByDate()).get_table_list(virtual_only=True)
+
+        for itable, vtable in enumerate(virtuals[::-1]):
+            table_key = dict(full_table_name=vtable['full_table_name'])
+
+            if Table & table_key:
+                dj.Table._update(self & table_key, 'table_order', itable)
+            else:
+                self.insert1(
+                    dict(
+                        full_table_name=vtable['full_table_name'],
+                        table_class=vtable['table'],
+                        table_order_category='virtual',
+                        table_order=itable,
+                        table_label='virtual'),
+                    skip_duplicates=True)
+
+    def insert_tables(self, table_type='All'):
+
+        if table_type == 'session':
+            self._insert_package_tables(SESSION_TABLES, table_type)
+        elif table_type == 'date':
+            self._insert_package_tables(DATE_TABLES, table_type)
+        elif table_type == 'virtual':
+            self._insert_virtual_tables()
+        elif table_type == 'All':
+            self._insert_package_tables(SESSION_TABLES, 'session')
+            self._insert_package_tables(DATE_TABLES, 'date')
+            self._insert_virtual_tables()
+        else:
+            ValueError('Invalid table_type. It has to be one of the following: session, date, virtual')
+
+
+table_kwargs = dict(order_by='table_order desc', as_dict=True)
+populate_kwargs = dict(
+    suppress_errors=True, display_progress=True,
+    return_exception_objects=True)
+
+
+@schema
+class Run(dj.Manual):
```

Comment:
Quite a cool logic but curious why you didn't turn this into a computed table?

Reply:
The reasons are:
1. I called a lot of populate in the make. It wouldn't work as a computed table.
2. I would like to track the status in real time, such as which table is populating, which ones are finished. I would want the transactions to happen in real time, not till the end of the program.

Reply:
Makes a lot of sense. Thanks for the clarification!